sdk/typescript/src/rpc/client.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import RpcClient from 'jayson/lib/client/browser';
+import {
+  literal,
+  type as pick,
+  string,
+  Struct,
+  unknown,
+  assert,
+  optional,
+  any,
+  is,
+} from 'superstruct';
+import fetch from 'cross-fetch';
+
+/**
+ * An object defining headers to be passed to the RPC server
+ */
+export type HttpHeaders = { [header: string]: string };
+
+export class JsonRpcClient {
+  private rpcClient: RpcClient;
+
+  constructor(url: string, httpHeaders?: HttpHeaders) {
+    this.rpcClient = this.createRpcClient(url, httpHeaders);
+  }
+
+  private createRpcClient(url: string, httpHeaders?: HttpHeaders): RpcClient {
```

Comment:
why is this a private method instead of a constructor ?

Reply:
This method is used in a constructor. I make it a separate function so that 1. make it clear that what this chunk of work is doing from the function name; 2. easier to reuse and unit test

Reply:
it should be just as easy to unit test and re-use as a constructor.  If we keep it as a method, it seems like it should at least be a static

Reply:
It's not easy to unit test anymore if the constructor contains other logic than `createRpcClient`. In general, we should prefer to define methods with specific names and single responsibility. But yeah I can make it static.